src/compiler/parser.rs

```diff
@@ -386,7 +397,11 @@ impl ParserType {
         self.set_parse_diagnostics(vec![]);
         self.set_parsing_context(ParsingContext::None);
 
-        self.scanner_mut().set_text(Some(_source_text), None, None);
+        let mut scanner = self.scanner_mut();
+        scanner.set_text(Some(_source_text), None, None);
+        // scanner.set_on_error(Some(Box::new(move |message, length| {
```

Comment:
The scanner in the Typescript version has an optional `onError` callback field which the parser initializes to call back into its `scanError()` function. I tried messing around with actually mimicking that (so storing a closure in the `Scanner` that captures a reference to the `ParserType` instance) but was running into lifetime stuff and I think conceptually it wouldn't make sense because if the scanner was effectively holding onto an immutable reference to the parser that should mean that the parser could never be used mutably while the scanner existed, which we don't want. So instead I used the same technique you suggested previously elsewhere of "injecting" the `on_error` callback when calling into scanner methods that might need to invoke it (so far just the one entry point from the parser of `scanner.scan()`)